card_game.py

```diff
@@ -0,0 +1,318 @@
+import pygame
+import sys
+import ai
+from pygame.locals import *
+
+"""The python code that everything runs from."""
+
+class DeckView(object):
+    """A view that shows where the deck is, and how many cards are left in it.
+
+        Attributes: visibility, model
+    """
+    def __init__(self,model):
+        self.model = model
+        self.visibility = True
+
+    def draw(self, surface):
+        """Used to describe how the deck is drawn in pygame."""
+        model = self.model
+        pygame.draw.rect(surface, game_constants.c_red, (int(model.x), int(model.y), model.width, model.height))
+
+        font = pygame.font.SysFont("monospace", 15)
+        deck_label = font.render(str(len(model.cards_in_deck)), 1, game_constants.c_black)
+        screen.blit(deck_label, (model.x, model.y))
+        if len(deck.cards_in_deck) > 0:         #As long as there are cards in the deck, it will be visible.
+            trump_card_val_label = font.render(str(model.cards_in_deck[0].value), 1, game_constants.c_black)
+            trump_card_suit_label = font.render(game_constants.suit_dict[model.cards_in_deck[0].suit], 1, game_constants.c_black)
+            pygame.draw.rect(surface, game_constants.c_white, (int(model.x) - game_constants.WIDTHCARD * 1.2, int(model.y), model.width, model.height))
+            screen.blit(trump_card_val_label, (model.x - game_constants.WIDTHCARD * 1.2, model.y))
+            screen.blit(trump_card_suit_label, (model.x - game_constants.WIDTHCARD * 1.2, model.y + game_constants.HEIGHTCARD * 2/3))
+        if len(deck.cards_in_deck) == 0:        #Auto-deletion of deck if there are no cards in it.
+            self.visibility = False
+
+    def delete(self):           #An alternative method to deleting the deck.
+        self.visibility = False
+
+class ButtonView(object):
+    """A view created for cards.
+
+    Attributes: model"""
+    def __init__(self, model):
+        self.model = model
+
+    def draw(self, surface):
+        model = self.model
+        pygame.draw.rect(surface, game_constants.c_blue, (int(model.x), int(model.y), model.width, model.height))
+
+class CardView(object):
+    """A view created for cards.
+
+    Attributes: model
+    """
+    def __init__(self, model):
+        self.model = model
+
+    def draw(self, surface):
+        model = self.model
+        pygame.draw.rect(surface, game_constants.c_white, (int(model.x), int(model.y), model.width, model.height))
+        if not model.opponent:      #if the card isn't in the opponent's hand it will show its value and suit.
+            font = pygame.font.SysFont("monospace", 15)
+            val_label = font.render(str(model.value), 1, game_constants.c_black)
+            suit_label = font.render(game_constants.suit_dict[model.suit], 1, game_constants.c_black)
+            screen.blit(val_label, (model.x, model.y))
+            screen.blit(suit_label, (model.x, model.y + game_constants.HEIGHTCARD * 2/3))
+
+class MoveController(object):
```

Comment:
Splitting files into Model View Controller might have made some sense, though honestly this seems cleaner for the scope of this project.